### ISSUE TITLE
Error when running npm run build on Windows

### ISSUE THREAD

**Issue by tsimbalar** [moarwick/elm-webpack-starter#19]
Hi,

I get an error when I try to build the default app for "distribution".

In the dist folder, I get some files, and the app actually runs, but the image is missing (and I guess that some other assets too)

Any idea what I am doing wrong ? 

Thanks in advance !

Below are the error details : 

```
'cp' n’est pas reconnu en tant que commande interne
ou externe, un programme exécutable ou un fichier de commandes.

npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run
" "build"
npm ERR! node v6.1.0
npm ERR! npm  v3.8.6
npm ERR! code ELIFECYCLE
npm ERR! elm-webpack-starter@0.6.1 build: `rimraf dist && webpack && cp -r src/img/ dist/img/ && cp src/favicon.ico dist
/favicon.ico`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the elm-webpack-starter@0.6.1 build script 'rimraf dist && webpack && cp -r src/img/ dist/img/ && cp
src/favicon.ico dist/favicon.ico'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the elm-webpack-starter package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     rimraf dist && webpack && cp -r src/img/ dist/img/ && cp src/favicon.ico dist/favicon.ico
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs elm-webpack-starter
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls elm-webpack-starter
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Dev\Perso\elm-webpack-starter\npm-debug.log
```

`npm-debug.log` contains the following : 

```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'build' ]
2 info using npm@3.8.6
3 info using node@v6.1.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle elm-webpack-starter@0.6.1~prebuild: elm-webpack-starter@0.6.1
6 silly lifecycle elm-webpack-starter@0.6.1~prebuild: no script for prebuild, continuing
7 info lifecycle elm-webpack-starter@0.6.1~build: elm-webpack-starter@0.6.1
8 verbose lifecycle elm-webpack-starter@0.6.1~build: unsafe-perm in lifecycle true
9 verbose lifecycle elm-webpack-starter@0.6.1~build: PATH: C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin;C:\Dev\Perso\elm-webpack-starter\node_modules\.bin;C:\Program Files\nodejs;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files\nodejs\;C:\WINDOWS\system32\config\systemprofile\.dnx\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Git\cmd;C:\Program Files (x86)\Skype\Phone\;C:\Program Files (x86)\Elm Platform\0.17\bin;C:\Users\tdesodt.DEV\AppData\Roaming\npm;C:\Program Files (x86)\Microsoft VS Code\bin
10 verbose lifecycle elm-webpack-starter@0.6.1~build: CWD: C:\Dev\Perso\elm-webpack-starter
11 silly lifecycle elm-webpack-starter@0.6.1~build: Args: [ '/d /s /c',
11 silly lifecycle   'rimraf dist && webpack && cp -r src/img/ dist/img/ && cp src/favicon.ico dist/favicon.ico' ]
12 silly lifecycle elm-webpack-starter@0.6.1~build: Returned: code: 1  signal: null
13 info lifecycle elm-webpack-starter@0.6.1~build: Failed to exec build script
14 verbose stack Error: elm-webpack-starter@0.6.1 build: `rimraf dist && webpack && cp -r src/img/ dist/img/ && cp src/favicon.ico dist/favicon.ico`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:239:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:191:7)
14 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:191:7)
14 verbose stack     at maybeClose (internal/child_process.js:850:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
15 verbose pkgid elm-webpack-starter@0.6.1
16 verbose cwd C:\Dev\Perso\elm-webpack-starter
17 error Windows_NT 10.0.10586
18 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "build"
19 error node v6.1.0
20 error npm  v3.8.6
21 error code ELIFECYCLE
22 error elm-webpack-starter@0.6.1 build: `rimraf dist && webpack && cp -r src/img/ dist/img/ && cp src/favicon.ico dist/favicon.ico`
22 error Exit status 1
23 error Failed at the elm-webpack-starter@0.6.1 build script 'rimraf dist && webpack && cp -r src/img/ dist/img/ && cp src/favicon.ico dist/favicon.ico'.
23 error Make sure you have the latest version of node.js and npm installed.
23 error If you do, this is most likely a problem with the elm-webpack-starter package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     rimraf dist && webpack && cp -r src/img/ dist/img/ && cp src/favicon.ico dist/favicon.ico
23 error You can get information on how to open an issue for this project with:
23 error     npm bugs elm-webpack-starter
23 error Or if that isn't available, you can get their info via:
23 error     npm owner ls elm-webpack-starter
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by MartinKavik** [comment]
Hi,

Windows doesn't know command **cp**, so the alternative for Win is:

```
"build": "rimraf dist && webpack && xcopy /S \"src/img/*\" \"dist/img/\" /F /Y && xcopy \"src/favicon.ico\" \"dist/\" /F /Y",
```

But it is ugly solution.
@moarwick: Is it possible to use something as [**copy-webpack-plugin**](https://www.npmjs.com/package/copy-webpack-plugin)? Thanks.

**Comment by moarwick** [comment]
yes, i've seen the `copy-webpack-plugin` out there, will give it a try as soon as i find some time.. in the meantime, your solution should help @tsimbalar (and other Win users) -- thanks!

**Comment by moarwick** [comment]
Ok, finally found a moment to look into implementing `copy-webpack-plugin`. I believe it's working so, if still interested, please try `0.6.2`. I don't have a Windows system handy so if you try it, pls let me know how it goes. Thanks!

**Comment by MartinKavik** [comment]
Nice! Everything seems to be working on Win.

@moarwick: Btw I want to build some projects on this repo in the near future and I have some ideas - e.g. integrate [webpack-notifier](https://www.npmjs.com/package/webpack-notifier) / show errors in the browser window. So I can help more If you want.

**Comment by kittykatattack** [comment]
Hi, I'm on Windows and would love to try this!
What's the best way to upgrade to the latest version without over-writing my current work?

**Comment by moarwick** [comment]
If you didn't modify your webpack configs or added any dependencies to package.json, just clone a new version of this repo, then copy the contents of your project's `src/` folder over, and.. it _should_ just work ;)

**Comment by kittykatattack** [comment]
@moarwick Thank you, it did work! :+1:

I also copied over my `elm-package.json` file and then ran `elm package install` which installed all the new packages I was using.
I also had to do a local re-install of a few npm packages I was using, like `marked.js`.
The only other thing I did was manually update the new version of `webpack.config.js` so that my custom folders in the `src` directory would be copied over into `dist`:

```
 plugins: [
      new CopyWebpackPlugin([
        {
          from: 'src/images/',
          to:   'images/'
        },
        {
          from: 'src/fonts/',
          to:   'fonts/'
        },
        {
          from: 'src/favicon.ico'
        },
      ]),
```

Your build fix for Windows works perfectly, by the way, thanks so much!
